source/agora/common/BlockStorage.d

```diff
@@ -110,6 +186,31 @@ public class BlockStorage
         return buildPath(this.path, format("B%012d.dat", index));
     }
 
+    /***************************************************************************
+
+        Returns:
+            the highest block
+
+    ***************************************************************************/
+
+    public const(Block)* getLastBlock () @safe nothrow
```

Comment:
Ok, I just realized that my comment about merging the commits may be wrong. However, I am not sure this function is actually safe.

This function was implemented as a workaround for the `Ledger` using a `static Block` as a temporary, right? But here we are again returning a pointer to a temporary which will be overwritten as soon as `getLastBlock` is called again.

I'll have to think about how we can make this better.

Reply:
Do you want me to do something?

Reply:
I am writing a proposal, I will let you know soon!